addons/html_builder/static/src/website_builder/plugins/options/dynamic_snippet_hook.js

```diff
@@ -0,0 +1,73 @@
+import { useDomState } from "@html_builder/core/utils";
+import { onWillStart, useState } from "@odoo/owl";
+
+export function useDynamicSnippetOption(
+    fetchDynamicFilters,
+    fetchDynamicFilterTemplates,
+    modelNameFilter,
+    contextualFilterDomain = []
+) {
+    onWillStart(async () => {
+        await fetchDynamicFiltersAndTemplates();
+    });
+    const state = useState({
+        defaultFilterId: undefined,
+        dynamicFilters: {}, // per id, to locate default filter
```

Comment:
`defaultFilterId` and `dynamicFilters` are not used outside this hook, I would not add it into the `cosnt state`.
As we discussed, we don't need useState, so `dynamicFilterTemplates` can directly be returned in like so:
```js
return { dynamicFilterTemplates, domState, getFilteredTemplates, showFilterOption };
```